File: Data Structures & Algorithms/jump-game/submission-4.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        farthest = 0

        for i in range(len(nums)):
            if farthest >= len(nums)-1:
                return True
            if i > farthest:
                return False
            farthest = max(farthest, i + nums[i])
        

            

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.empty()){
            return 0;
        }
        vector<int> sorted_nums = nums;
        sort(sorted_nums.begin(), sorted_nums.end());
        vector<int> maxes;
        int count = 1;
        
        for(int i =1;i<sorted_nums.size();++i){
            if(sorted_nums[i-1]+1==sorted_nums[i]) {
                ++count;
            }
            else if(sorted_nums[i-1]==sorted_nums[i]){
                continue;
            }
            else{
                maxes.push_back(count);
                count = 1;
            }
            maxes.push_back(count);
        }
        int max = 1;
        for(int i = 0;i<maxes.size();++i){
            if(maxes[i]>max){
                max = maxes[i];
            }
        }
        return max;
    }



};